presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -91,11 +91,15 @@ public CursorProcessor compileCursorProcessor(RowExpression filter, List<RowExpr
         }
     }
 
-    public PageProcessor compilePageProcessor(RowExpression filter, List<RowExpression> projections)
+    public Class<? extends PageProcessor> compilePageProcessor(RowExpression filter, List<RowExpression> projections)
```

Comment:
Can we make this return `Supplier<PageProcessor>`?  I think that would make it easier for callers and might make it more clear that they should be creating multiple instances